test/test_type_promotion.py

```diff
@@ -959,7 +959,14 @@ def test_computation_ignores_out(self, device):
         self.assertNotEqual(result, a.double() - b, exact_dtype=False)
 
     def test_atan2_type_promotion(self, device):
-        dtypes = torch.testing.get_all_fp_dtypes(include_half=True)
+        include_half = False
+        if self.device_type == 'cuda':
+            # Half is not supported on CPU.
+            include_half = True
+
+        dtypes = torch.testing.get_all_fp_dtypes(include_half=include_half, include_bfloat16=False) + \
```

Comment:
You can use the `@dtypes` decorator with pairs of dtypes if you like. For example:

```
@dtypes(*product(torch.testing.get_all_dtypes(), torch.testing.get_all_dtypes())
def test_atan2_type_promotion(self, device, dtypes):
  dtype1, dtype2 = dtypes
  ...
```

Reply:
Actually it will be hard to use as include_half is chosen based on device type.

Reply:
You can use 

```
@dtypesIfCUDA(...)
@dtypes(...)
```

CUDA will get the dtypes in the first decorator, all other device types will get the dtypes in the second decorator. If you want CPU-specific dtypes there's also `@dtypesIfCPU`.